File: seed/ruby-sdk-v2/trace/lib/fern_trace/submission/types/test_case_result_with_stdout.rb
# frozen_string_literal: true

module FernTrace
  module Submission
    module Types
      class TestCaseResultWithStdout < Internal::Types::Model
        field :result, -> { FernTrace::Submission::Types::TestCaseResult }, optional: false, nullable: false
        field :stdout, -> { String }, optional: false, nullable: false
      end
    end
  end
end

File: seed/pydantic-v2/trace/baseline/resources/submission/types/test_case_result_with_stdout.py
from pydantic import BaseModel
from resources.submission.types.test_case_result import TestCaseResult
from dt import datetime
from core.datetime_utils import serialize_datetime
class TestCaseResultWithStdout(BaseModel):
    result: TestCaseResult
    stdout: str
    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

